docs/debug_faq.md

```diff
@@ -0,0 +1,177 @@
+debugging and adding context to bug reports. They look like pretty much just
+like Tilt, but frozen in time:
+
+[https://cloud.tilt.dev/snapshot/AYSV59gLhM3GVMuuR28=](https://cloud.tilt.dev/snapshot/AYSV59gLhM3GVMuuR28=)
+
+A snapshot is a frozen "moment-in-time" version of the Tilt UI. In a snapshot
+you can drill in to specific services, see alerts and Kubernetes events. Pretty
+much anything you can do in normal Tilt, you can do in a snapshot!
+
+This is also helpful to send along with bug reports.
+
+For more information, see "[Share Errors and Cluster State with Snapshots](snapshots.html)".
+
+---
+
+### Why does my image build fail in Tilt when it succeeds with `docker build`?
+
+Run:
+
+```shell
+tilt docker -- build ARGS
+```
+
+This will run Docker the same way that Tilt runs Docker.
+
+Tilt automatically enables optimizations that you may not be using by
+default. For example, if you are using minikube and run:
+
+```shell
+tilt docker -- build -t image-name .
+```
+
+Tilt may print:
+
+```
+Running Docker command as:
+DOCKER_HOST=tcp://192.168.99.100:2376 DOCKER_CERT_PATH=/home/nick/.minikube/certs DOCKER_TLS_VERIFY=1 docker build -t image-name .
+```
+
+because it's running against Minikube's Docker instance, not your Docker instance.
+
+---
+
+### Why is Tilt using so much CPU or memory?
+
+Please file an issue if Tilt is being a resource hog! We've made it a lot better
+in the last few months but there may still be builds that cause problems.
+
+As of v0.10.26, Tilt exposes the standard Go pprof hooks over [HTTP](https://golang.org/pkg/net/http/pprof/).
+
+To look at a 30-second CPU profile:
+
+```shell
+go tool pprof http://localhost:10350/debug/pprof/profile?seconds=30
+```
+
+To look at the heap profile:
+
+```shell
+go tool pprof http://localhost:10350/debug/pprof/heap
+```
+
+This opens a special REPL that lets you explore the data.
+Type `web` in the REPL to see a CPU graph.
+
+For more information on pprof, see [the Go pprof guide](https://github.com/google/pprof/blob/master/doc/README.md).
+
+---
+
+### What does Tilt think is happening right now?
+
+The internal Tilt engine is implemented as a control loop, just like Kubernetes.
+
+Tilt watches lots of different inputs (your Tiltfile, your local source files,
+and your Kubernetes cluster) for changes. The control loop records these
+changes in a central state store. Then, Tilt kicks off updates to your cluster
+based on the state store.
+
+That means all the state Tilt knows about lives in a single place. And you can inspect it!
+
+While Tilt is running in one terminal, open another terminal and run:
+
+```shell
+tilt dump engine
+```
+
+Tilt will print a JSON representation of everything it knows about your build state and your cluster state.
+
+The Tilt UI has a similar control loop. Run:
+
+```shell
+tilt dump webview
+```
+
+to see the complete state of the Tilt web UI.
+
+---
+
+### How can I monitor Tilt usage on my team?
```

Comment:
Do we want to document this?

Reply:
if people end up using it based on minimal documentation, that would be a great outcome!